src/radical/pilot/pilot.py

```diff
@@ -654,15 +654,23 @@ def stage_in(self, sds):
 
         for sd in sds:
             sd['prof_id'] = self.uid
-
-        for sd in sds:
             sd['source'] = str(complete_url(sd['source'], self._loc_ctx, self._log))
-            sd['target'] = str(complete_url(sd['target'], self._rem_ctx, self._log))
 
-        # ask the pmgr to send the staging reuests to the stager
+            # case when target url is absolute path
+            if sd['target'].startswith('pilot:///'):
```

Comment:
I think this means that (instead of addressing absolute paths in the target environment, directly), this adds explicit handling for the `pilot://` URI scheme with a null "authority" component?

And, otherwise, plain absolute paths are converted to `file:///` paths, which always refer to the client side?

I think this warrants an update of the notes at https://radicalpilot.readthedocs.io/en/stable/datastaging.html#staging-directives, too.

Reply:
If you *do* want to implement a URI scheme that places files to anywhere in the remote filesystem, a better scheme name would be `resource` or `endpoint`. A URI like `endpoint://resourcename/` as an alias to the filesystem endpoint URI stub in the `resource_X.json` file would be pretty cool. But it isn't what issue #2590 was asking for.

Reply:
> If you _do_ want to implement a URI scheme that places files to anywhere in the remote filesystem, a better scheme name would be `resource` or `endpoint`. A URI like `endpoint://resourcename/` as an alias to the filesystem endpoint URI stub in the `resource_X.json` file would be pretty cool. But it isn't what issue #2590 was asking for.

That sounds better, and that's similar to what I and @andre-merzky discussed.
I have made a new URL `scheme` (access protocol name) -->> `pilot_fs://` which points to filesystem_endpoint configuration of pilot specified in resource_X.json. Have a look.